contrib/mesos/pkg/executor/executor.go

```diff
@@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
 	// (a) we ignore future launches for it
 	// (b) we have a record of it so that we can kill it if needed
 	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
+	var launchTimer *time.Timer
+	if k.launchGracePeriod > 0 {
+		launchTimer = time.NewTimer(k.launchGracePeriod)
+	} else {
+		launchTimer = &time.Timer{}
```

Comment:
not better. then i'm checking launchTimer == nil in places. i'd just rather
be able to refer to launchTimer.C and know that i'm not going to trigger a
panic

On Wed, Dec 9, 2015 at 11:32 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/17917#discussion_r47114263>
> :
>
> > @@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
> >  	// (a) we ignore future launches for it
> >  	// (b) we have a record of it so that we can kill it if needed
> >  	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
> > +	var launchTimer *time.Timer
> > +	if k.launchGracePeriod > 0 {
> > +		launchTimer = time.NewTimer(k.launchGracePeriod)
> > +	} else {
> > +		launchTimer = &time.Timer{}
>
> Why not launchTimer == nil then? Much better than an empty struct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17917/files#r47114263>.
>